x/trigger/client/cli/tx.go

```diff
@@ -0,0 +1,261 @@
+		},
+	}
+	flags.AddTxFlagsToCmd(cmd)
+	return cmd
+}
+
+// GetCmdDestroyTrigger is a command to destroy an existing trigger.
+func GetCmdDestroyTrigger() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     "destroy-trigger [id]",
+		Args:    cobra.ExactArgs(1),
+		Aliases: []string{"destroy", "d"},
+		Short:   "Destroys an existing trigger.",
+		Long:    strings.TrimSpace(`Destroys an existing trigger. The trigger will not be destroyable if it has already been detected.`),
+		Example: fmt.Sprintf(`$ %[1]s tx trigger destroy-trigger 1`, version.AppName),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			clientCtx, err := client.GetClientTxContext(cmd)
+			if err != nil {
+				return err
+			}
+			callerAddr := clientCtx.GetFromAddress()
+			triggerID, err := strconv.Atoi(args[0])
+			if err != nil {
+				return fmt.Errorf("invalid trigger id: %s", args[0])
+			}
+
+			msg := types.NewDestroyTriggerRequest(
+				callerAddr.String(),
+				uint64(triggerID),
+			)
+			return tx.GenerateOrBroadcastTxCLI(clientCtx, cmd.Flags(), msg)
+		},
+	}
+	flags.AddTxFlagsToCmd(cmd)
+	return cmd
+}
+
+// parseTransactions reads and parses the message.
+func parseTransactions(cdc codec.Codec, path string) ([]sdk.Msg, error) {
+	contents, err := os.ReadFile(path)
+	if err != nil {
+		return nil, err
+	}
+
+	var action ParsedAction
+	err = json.Unmarshal(contents, &action)
+	if err != nil {
+		return nil, err
+	}
+
+	var msg sdk.Msg
+	err = cdc.UnmarshalInterfaceJSON(action.Action, &msg)
+	if err != nil {
+		return nil, err
+	}
+
+	return []sdk.Msg{msg}, nil
+}
+
+// parseEvent reads and parses the transaction event from a file.
+func parseEvent(path string) (*types.TransactionEvent, error) {
+	contents, err := os.ReadFile(path)
+	if err != nil {
+		return nil, err
+	}
+
+	var event ParsedTransactionEvent
+	err = json.Unmarshal(contents, &event)
+	if err != nil {
+		return nil, err
+	}
+
+	newEvent := types.TransactionEvent{
+		Name: event.Name,
+	}
+	for _, attr := range event.Attributes {
+		newEvent.Attributes = append(newEvent.Attributes, types.Attribute{
+			Name:  attr.Name,
+			Value: attr.Value,
+		})
+	}
+
+	return &newEvent, nil
+}
+
+// ParsedAction is the deserialized form of the inputted JSON formatted sdk.Msg.
+type ParsedAction struct {
+	// Action defines a sdk.Msg proto-JSON-encoded as Any.
+	Action json.RawMessage `json:"message,omitempty"`
+}
+
+// ParsedTransactionEvent is the deserialized form of the inputted JSON formatted transaction event.
+type ParsedTransactionEvent struct {
+	// Attributes defines an array of ParsedAttribute.
+	Attributes []ParsedAttribute `json:"attributes,omitempty"`
+	// The name/type of the event.
+	Name string `json:"name"`
+}
+
+// ParsedTransactionEvent is the deserialized form of the inputted JSON formatted transaction event's attribute.
+type ParsedAttribute struct {
+	// The name of the attribute.
+	Name string `json:"name"`
+	// The value of the attribute.
+	Value string `json:"value"`
+}
```

Comment:
Is there a reason you're not using the `TransactionEvent` and `Attribute` directly here?

I'd hate to end up in a situation similar to the upgrade module where they define things one way for CLI parsing, and that way is different from what's defined in the protos. E.g., you can't serialize an upgrade to JSON then provide it to the create-upgrade command because that command wants different JSON fields.

Reply:
Let me look into this. It was primarily done this way because I was following a cosmos example. It would be nice to remove these redundant structures.